### PR TITLE
[14.0] [mig] pos_invoicing

### PR DESCRIPTION
**Pull request by fshah-initos**

### CODE REVIEW

File: pos_invoicing/__manifest__.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 {
     "name": "Point Of Sale - Invoicing",
     "summary": "Handle invoicing from Point Of Sale",
-    "version": "12.0.3.0.1",
+    "version": "14.0.0.1.0",
```
Comment (ivantodorovich) [inline pos_invoicing/__manifest__.py]:
```suggestion
    "version": "14.0.1.0.0",
```

File: pos_invoicing/models/account_move.py [hunk 2]
```diff
@@ -3,37 +3,33 @@
...
         self._check_pos_pending_payment()
-        return super().action_cancel()
+        return super().button_cancel()
 
-    @api.one
     def _get_outstanding_info_JSON(self):
         if self.pos_pending_payment:
             return
         else:
             return super()._get_outstanding_info_JSON()
```
Comment (ivantodorovich) [inline pos_invoicing/models/account_move.py]: This method doesn't exist in upstream anymore, AFAICS

**Review by ivantodorovich — CHANGES_REQUESTED** [review]
Thanks! left some comments :)
Also please check pre-commit

**Comment by fshah-initos** [comment]
Hi @ivantodorovich
Can you please review the checks that are passed.

File: pos_invoicing/models/pos_session.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Copyright (C) 2013 - Today: GRAP (http://www.grap.coop)
+# @author: Julien WESTE
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import models
+
+_logger = logging.getLogger(__name__)
```
Comment (ivantodorovich) [inline pos_invoicing/models/pos_session.py]: You can remove `_logger` from this file, as it isn't used

File: pos_invoicing/models/pos_session.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class PosSession(models.Model):
+    _inherit = "pos.session"
+
+    def action_pos_session_close(self):
+        res = super().action_pos_session_close()
+        orders = self.mapped("order_ids").filtered(lambda x: x.account_move)
+        orders.mapped("account_move").write({"pos_pending_payment": False})
```
Comment (ivantodorovich) [inline pos_invoicing/models/pos_session.py]:
```suggestion
        self.order_ids.account_move.pos_pending_payment = False
```

This is the same and more readable probably

File: pos_invoicing/models/account_move.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    pos_pending_payment = fields.Boolean(
+        string="PoS - Pending Payment",
+        readonly=True,
+        copy=False,
+        help="Indicates an invoice for which there are pending payments in the"
+        " Point of Sale. \nThe invoice will be marked as paid when the session"
+        " will be closed.",
+    )
+
+    # Overload Section
```
Comment (ivantodorovich) [inline pos_invoicing/models/account_move.py]:
```suggestion
```

File: pos_invoicing/models/account_move.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    def button_cancel(self):
+        self._check_pos_pending_payment()
+        return super().button_cancel()
+
+    def _check_pos_pending_payment(self):
+        invoices = self.filtered(lambda x: x.pos_pending_payment)
```
Comment (ivantodorovich) [inline pos_invoicing/models/account_move.py]:
```suggestion
        invoices = self.filtered("pos_pending_payment")
```
same & more readable than lambda

File: pos_invoicing/models/account_move.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        string="PoS - Pending Payment",
+        readonly=True,
+        copy=False,
+        help="Indicates an invoice for which there are pending payments in the"
+        " Point of Sale. \nThe invoice will be marked as paid when the session"
+        " will be closed.",
+    )
+
+    # Overload Section
+    def button_cancel(self):
```
Comment (ivantodorovich) [inline pos_invoicing/models/account_move.py:22]: Can you add a test case for this?
Please use `assertRaisesRegex` to check that the right exception is raised

File: pos_invoicing/models/account_payment.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = "account.payment.register"
+
+    def action_create_payments(self):
+        context = self.env.context
+        active_id = context.get("active_id")
+        account_move = self.env["account.move"].browse(active_id)
+        if account_move.pos_pending_payment:
+            raise UserError(
```
Comment (ivantodorovich) [inline pos_invoicing/models/account_payment.py:17]: Can you add a test case for this?

File: pos_invoicing/models/account_payment.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+                    "You can not realize this action on the payments(s) %s because"
+                    " there are pending payments in the Point of Sale."
+                )
+                % (", ".join(account_move.mapped("name")))
+            )
+        return super().action_create_payments()
+
+    def action_post(self):
+        if self.move_id.pos_pending_payment:
+            raise UserError(
```
Comment (ivantodorovich) [inline pos_invoicing/models/account_payment.py]: Same, could you add a test case for this too?

File: pos_invoicing/tests/test_module.py [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+# @author: Julien WESTE
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields
+from odoo.tests.common import TransactionCase
+
+
+class TestModule(TransactionCase):
+    def setUp(self):
```
Comment (ivantodorovich) [inline pos_invoicing/tests/test_module.py]: Please do this in `setUpClass` instead.

File: pos_invoicing/tests/test_module.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+        # Get Object
+        self.pos_product = self.env.ref("point_of_sale.whiteboard_pen")
+        self.pricelist = self.env.ref("product.list0")
+        self.partner = self.env.ref("base.res_partner_12")
+
+        # Create a new pos config and open it
+        self.pos_config = self.env.ref("point_of_sale.pos_config_main").copy()
+        self.pos_config.open_session_cb()
+
+    # Test Section
```
Comment (ivantodorovich) [inline pos_invoicing/tests/test_module.py]:
```suggestion
```
duh!

File: pos_invoicing/tests/test_module.py [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+        order.action_pos_order_invoice()
+
+        self.assertEqual(order.account_move.pos_pending_payment, True)
+
+        # Once closed check if the invoice is correctly set
+        self.pos_config.current_session_id.action_pos_session_closing_control()
+        self.assertEqual(order.account_move.pos_pending_payment, False)
+
+    def _create_order(self):
```
Comment (ivantodorovich) [inline pos_invoicing/tests/test_module.py:62]: Move this helper up, right after `setUpClass`

**Review by ivantodorovich — COMMENTED** [review]
Thanks @fshah-initos !

Code coverage is still pretty low, do you think you could add some more test cases?
I'm not blocking on this, but it'd be really good if you do :)

Thanks!

File: pos_invoicing/models/pos_session.py (hunk 3 above)
Comment (ikapasi-initos) [reply pos_invoicing/models/pos_session.py]: Already removed _logger in this file.

File: pos_invoicing/models/pos_session.py (hunk 4 above)
Comment (ikapasi-initos) [reply pos_invoicing/models/pos_session.py]: This change is already implemented can you check it?

**Comment by ikapasi-initos** [comment]
@ivantodorovich Can you please review again this pull?

**Comment by dsolanki-initos** [comment]
Maintainers team, Could you please reopen this PR?

**Review by ivantodorovich — APPROVED** [review]

**Pull request closed without merging**